src/conversions/chrono.rs

```diff
@@ -0,0 +1,848 @@
+        let dt = <PyDateTime as PyTryFrom>::try_from(ob)?;
+        let ms = dt.get_fold() as u32 * 1_000_000 + dt.get_microsecond();
+        let h = dt.get_hour().into();
+        let m = dt.get_minute().into();
+        let s = dt.get_second().into();
+        let tz = if let Some(tzinfo) = dt.get_tzinfo() {
+            tzinfo.extract()?
+        } else {
+            return Err(PyTypeError::new_err("Not datetime.timezone.tzinfo"));
+        };
+        let dt = NaiveDateTime::new(
+            NaiveDate::from_ymd(dt.get_year(), dt.get_month().into(), dt.get_day().into()),
+            NaiveTime::from_hms_micro(h, m, s, ms),
+        );
+        Ok(DateTime::from_utc(dt, tz))
+    }
+}
+
+impl FromPyObject<'_> for DateTime<Utc> {
+    fn extract(ob: &PyAny) -> PyResult<DateTime<Utc>> {
+        let dt = <PyDateTime as PyTryFrom>::try_from(ob)?;
+        let ms = dt.get_fold() as u32 * 1_000_000 + dt.get_microsecond();
+        let h = dt.get_hour().into();
+        let m = dt.get_minute().into();
+        let s = dt.get_second().into();
+        let tz = if let Some(tzinfo) = dt.get_tzinfo() {
+            tzinfo.extract()?
+        } else {
+            return Err(PyTypeError::new_err("Not datetime.timezone.utc"));
+        };
+        let dt = NaiveDateTime::new(
+            NaiveDate::from_ymd(dt.get_year(), dt.get_month().into(), dt.get_day().into()),
+            NaiveTime::from_hms_micro(h, m, s, ms),
+        );
+        Ok(DateTime::from_utc(dt, tz))
+    }
+}
+
+// Utiliy function used to convert PyDelta to timezone
+fn pytimezone_fromoffset<'a>(py: &Python<'a>, td: &PyDelta) -> &'a PyAny {
+    // Safety: py.from_borrowed_ptr needs the cast to be valid.
+    // Since we are forcing a &PyDelta as input, the cast should always be valid.
+    unsafe {
+        PyDateTime_IMPORT();
+        py.from_borrowed_ptr(PyTimeZone_FromOffset(td.as_ptr()))
+    }
+}
+
+impl ToPyObject for FixedOffset {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        let seconds_offset = self.local_minus_utc();
+        // XXX: Here we don't normalize, otherwise the meaning
+        // of the offset changes. Normalizing a negative PyDelta and converting
+        // back to rust would transform a `-00:00:01` offset into `23:59:59`.
+        let td =
+            PyDelta::new(py, 0, seconds_offset, 0, false).expect("Failed to contruct timedelta");
+        pytimezone_fromoffset(&py, td).into()
+    }
+}
+
+impl IntoPy<PyObject> for FixedOffset {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        ToPyObject::to_object(&self, py)
+    }
+}
+
+impl FromPyObject<'_> for FixedOffset {
+    /// Convert python tzinfo to rust [`FixedOffset`].
+    ///
+    /// Note that the conversion will result in precision lost in microseconds as chrono offset
+    /// does not supports microseconds.
+    fn extract(ob: &PyAny) -> PyResult<FixedOffset> {
+        let py_tzinfo = <PyTzInfo as PyTryFrom>::try_from(ob)?;
```

Comment:
@pickfire this is my point - here it looks like we accept any tzinfo and convert it to a fixed offset. If we are going for a no-conversions approach we might need to be careful here? E.g. only accept `datetime.timezone` objects?

Reply:
I think this part is fine because fixedoffset can be anything, but utc should always be utc I think, hence it has a special value for utc. At most, we lost utc when converting back for this part.

Reply:
Yeah, given that a Python `tzinfo` might not represent a fixed offset, it seems that we should only accept a `datetime.timezone` here.

Reply:
Fair enough.